en/docs/InstallAndSetup/PerfromanceTuningAndTestResults/tuning-performance.md

```diff
@@ -81,41 +81,54 @@ In a clustered environment, the entity expansion limit has no dependency on the
 
 In multitenant mode, the WSO2 Carbon runtime limits the thread execution time. That is, if a thread is stuck or taking a long time to process, Carbon detects such threads, interrupts and stops them. Note that Carbon prints the current stack trace before interrupting the thread. This mechanism is implemented as an Apache Tomcat valve. Therefore, it should be configured in the `<PRODUCT_HOME>/repository/conf/tomcat/catalina-server.xml` file as shown below.
 
+!!! danger
+    Cannot configure using deployment.toml
 ``` xml
-    <Valve className="org.wso2.carbon.tomcat.ext.valves.CarbonStuckThreadDetectionValve" threshold="600"/>
+<Valve className="org.wso2.carbon.tomcat.ext.valves.CarbonStuckThreadDetectionValve" threshold="600"/>
 ```
 
 -   The `className` is the Java class used for the implementation. Set it to `org.wso2.carbon.tomcat.ext.valves.CarbonStuckThreadDetectionValve` .
 -   The `threshold` gives the minimum duration in seconds after which a thread is considered stuck. The default value is 600 seconds.
 
 #### APIM-level settings
 
--   [Timeout configurations for an API call](#TuningPerformance-TimeoutconfigurationsforanAPIcall)
--   [General APIM-level recommendations](#TuningPerformance-GeneralAPIM-levelrecommendations)
--   [Registry indexing configurations](#TuningPerformance-Registryindexingconfigurations)
--   [Tuning data-agent parameters](#TuningPerformance-Tuningdata-agentparameters)
+-   [Timeout configurations for an API call](#timeout-configurations-for-an-api-call)
+-   [General APIM-level recommendations](#general-apim-level-recommendations)
+-   [Registry indexing configurations](#registry-indexing-configurations)
+-   [Tuning data-agent parameters](#tuning-data-agent-parameters)
 
 ##### Timeout configurations for an API call
 
 The following diagram shows the communication/network paths that occur when an API is called. The timeout configurations for each network call are explained below.
-![]({{base_path}}/assets/attachments/103333648/103333650.png)
+
+![](../../../../assets/attachments/103333648/103333650.png)
+
+!!! info
+    Gateway to Keymanager network call to validate token only happens with the OAuth token. This network call does not happen for JWT tokens . From WSO2 API Manager 3.0.0 onwards, JWT tokens are the default token type for applications. Since JWT tokens are self-contained access tokens, Keymanager is not needed to validate the token. The token is validated from the gateway.
```

Comment:
- Keymanager should be Key Manager
- There is an extra space after JWT tokens and the period (.)

I reworded the sentence based on the above two reasons and also to improve the grammar. Therefore, shall we reword this to the following:

The Gateway to Key Manager network call to validate the token only happens with the OAuth token. This network call does not happen for JWT tokens. From WSO2 API Manager 3.0.0 onwards, JWT tokens are the default token type for applications. As JWT tokens are self-contained access tokens, the Key Manager is not needed to validate the token and the token is validated from the Gateway.

Reply:
updated